extension-manifest-v3/src/utils/engines.js

```diff
@@ -92,12 +92,27 @@ async function getDB() {
 
 async function loadFromStorage(name) {
   try {
-    const db = await getDB();
-
-    const tx = db.transaction('engines');
-    const table = tx.objectStore('engines');
-
-    const engineBytes = await table.get(name);
+    const engineBytes = await getDB()
```

Comment:
Perhaps using async-syntax would make it slightly easier to read than promise chaining? Just an opinion:

```
  try {
    const db = await getDB();
    const tx = db.transaction('engines');
    const table = tx.objectStore('engines');
    const result = await table.get(name);
    await tx.done;
    return result;
  } catch (e) {
    // Suppress the private browsing mode in Firefox
    if (!e.message?.includes('database that did not allow mutations')) {
      captureException(e);
    }

    if (__PLATFORM__ === 'firefox') {
      const key = `engines:${name}`;
      const data = await chrome.storage.local.get([key]);
      return data[key];
    } else {
      throw e;
    }
  }
  ```

Reply:
We still needs a `try/catch` around the deserialization - for example, it throws for adblocker version mismatch - I am not sure about nested try/catch...

Reply:
Alright. I have no strong opinion either.